python/lsst/validate/drp/checkAstrometryPhotometry.py

```diff
@@ -0,0 +1,514 @@
+    return ra, dec
+
+
+def magNormDiff(cat):
+    """Calculate the normalized mag/mag_err difference from the mean for a 
+    set of observations of an objection.
+
+    Inputs
+    ------
+    cat : collection with a .get method 
+         for flux, flux+"-" 
+
+    Returns
+    -------
+    pos_median : float
+        median diff of positions in milliarcsec.
+    """
+    mag = cat.get('base_PsfFlux_mag')
+    magerr = cat.get('base_PsfFlux_magerr')
+    mag_avg = np.mean(mag)
+    N = len(mag)
+    normDiff = (mag - mag_avg) / magerr
+    
+
+def positionRms(cat):
+    """Calculate the RMS for RA, Dec for a set of observations an object.
+
+    Inputs
+    ------
+    cat -- collection with a .get method 
+         for 'coord_ra', 'coord_dec' that returns radians.
+
+    Returns
+    -------
+    pos_rms -- RMS of positions in milliarcsecond.  Float.
+
+    This routine doesn't handle wrap-around
+    """
+    ra_avg, dec_avg = averageRaDec(cat)
+    ra, dec = cat.get('coord_ra'), cat.get('coord_dec')
+    # Approximating that the cos(dec) term is roughly the same
+    #   for all observations of this object.
+    ra_var = np.var(ra) * np.cos(dec_avg)**2
+    dec_var = np.var(dec)
+    pos_rms = np.sqrt(ra_var + dec_var)  # radians
+    pos_rms = afwGeom.radToMas(pos_rms)  # milliarcsec
+
+    return pos_rms
+
+
+def loadAndMatchData(repo, visitDataIds,
+                     matchRadius=afwGeom.Angle(1, afwGeom.arcseconds)):
+    """Load data from specific visit.  Match with reference.
+
+    Inputs
+    ------
+    repo : string
+        The repository.  This is generally the directory on disk
+        that contains the repository and mapper.
+    visitDataIds : list of dict
+        List of `butler` data IDs of Image catalogs to compare to reference.
+        The `calexp` cpixel image is needed for the photometric calibration.
+    matchRadius :  afwGeom.Angle().
+        Radius for matching. 
+
+    Returns
+    -------
+    afw.table.GroupView
+        An object of matched catalog.
+    """
+
+    # Following 
+    # https://github.com/lsst/afw/blob/tickets/DM-3896/examples/repeatability.ipynb
+    butler = dafPersist.Butler(repo)
+    dataset = 'src'
+
+    dataRefs = [dafPersist.ButlerDataRef(butler, vId) for vId in visitDataIds]
+
+    ccdKeyName = getCcdKeyName(visitDataIds[0])
+
+    schema = butler.get(dataset + "_schema", immediate=True).schema
+    mapper = SchemaMapper(schema)
+    mapper.addMinimalSchema(schema)
+    mapper.addOutputField(Field[float]('base_PsfFlux_mag', "PSF magnitude"))
+    mapper.addOutputField(Field[float]('base_PsfFlux_magerr', "PSF magnitude uncertainty"))
+    newSchema = mapper.getOutputSchema()
+
+    # Create an object that can match multiple catalogs with the same schema
+    mmatch = MultiMatch(newSchema, 
+                        dataIdFormat = {'visit': int, ccdKeyName: int},
+                        radius=matchRadius,
+                        RecordClass=SimpleRecord)
```

Comment:
Should `RecordClass` be `SourceRecord`? What you have seems to work ok, in fairness, but `SourceRecord` seems correct according to the `MultiMatch` docs.